content/200-concepts/100-components/02-prisma-client/056-composite-types.mdx

```diff
@@ -687,3 +705,90 @@ const orders = await prisma.order.findMany({
   },
 })
 ```
+
+## Duplicate values in unique fields of composite types
+
+Be careful when you carry out any of the following operations on a record with a composite type that has a unique constraint. In this situation, MongoDB does not enforce unique values inside a record.
+
+- When you create the record
+- When you add data to the record
+- When you update data in the record
+
+If your schema has a composite type with a `@@unique` constraint, MongoDB prevents you from storing the same value for the constrained value in two or more of the records that contain this composite type. However, MongoDB does does not prevent you from storing multiple copies of the same field value in a single record.
+
+Note that you can [use Prisma relations to work around this issue](#use-prisma-relations-to-enforce-unique-values-in-a-record).
+
+For example, in the following schema, `MailBox` has a composite type, `addresses`, which has a `@@unique` constraint on the `email` field.
+
+```prisma
+type Address {
+  email String
+}
+
+model MailBox {
+  name       String
+  addresses  Address[]
+
+  @@unique([addresses.email])
+}
+```
+
+The following code creates a record with two identical values in `address`. MongoDB does not throw an error in this situation, and it stores `alice@prisma.io` in `addresses` twice.
+
+```ts
+await prisma.MailBox.createMany({
+  data: [
+    {
+      name: 'Alice',
+      addresses: {
+        set: [
+          {
+            address: 'alice@prisma.io',
+          },
+          {
+            address: 'alice@prisma.io',
+          },
+        ],
+      },
+    },
+  ],
+})
+```
+
+Note: MongoDB throws an error if you try to store the same value in two separate records. In our example above, if you try to store the email address `alice@prisma.io` for the user Alice and for the user Bob, MongoDB does not store the data and throws an error.
+
+### Use Prisma relations to enforce unique values in a record
+
+In the example above, MongoDB did not enforce the unique constraint on a nested address name. However, you can structure your data differently to enforce unique values in a record. To do so, use Prisma [relations](/concepts/components/prisma-schema/relations) to turn the composite type into a collection. Set a relationship to this collection and place a unique constraint on the field that you want to be unique.
+
+In the following example, MongoDB enforces unique values in a record. There is a relation between the `Mailbox.addresses` composite type and the `Address` model, and he `name` field in the `Address` model has a unique constraint.
```

Comment:
```suggestion
In the following example, MongoDB enforces unique values in a record. There is a relation between the `Mailbox` and the `Address` model,  and the `name` field in the `Address` model has a unique constraint.
```

`Mailbox.addresses` is no longer considered a composite type as it's now refering to a new collection.

Reply:
Thanks - I'm going to accept this suggestion with a slight wording change